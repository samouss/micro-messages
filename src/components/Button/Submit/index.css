.Submit {
  background: #0169d9;
  border: 1px solid #225378;
  color: #FFF;
  border-radius: 3px;
  padding: 5px 10px;
  text-transform: uppercase;
  font-weight: 600;
  font-size: 0.85rem;
}
