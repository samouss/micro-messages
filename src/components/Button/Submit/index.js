// @flow

import React from 'react';
import './index.css';

export type Props = {
  children: React.Element<*>,
};

const Submit = ({ children, ...props }: Props) => (
  <button
    {...props}
    styleName="Submit"
  >
    {children}
  </button>
);

export default Submit;
