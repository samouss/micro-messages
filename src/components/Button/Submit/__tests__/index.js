// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Submit from '../index';

describe('<Submit />', () => {
  const defaultProps = {
    onClick: () => {},
  };

  it('expect to render the component correctly', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Submit
        {...props}
      >
        <span>some content</span>
      </Submit>,
    );

    expect(component).toMatchSnapshot();
  });
});
