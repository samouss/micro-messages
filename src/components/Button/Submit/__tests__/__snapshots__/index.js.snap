// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Submit /> expect to render the component correctly 1`] = `
<button
  onClick={[Function]}
  styleName="Submit"
>
  <span>
    some content
  </span>
</button>
`;
